### PR TITLE
[VPN] Refactoring IPSec policy resource

### LINKED ISSUE #640: [VPN] Missing support for latest PFS groups
# Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform,
please upgrade because your issue may have already been fixed.

> Terraform v0.12.26
> + provider.opentelekomcloud v1.20.2 

# Affected Resource(s)
Please list the resources as a list, for example:
- opentelekomcloud_vpnaas_ipsec_policy_v2
- opentelekomcloud_vpnaas_ike_policy_v2

# Terraform Configuration Files
```hcl
resource "opentelekomcloud_vpnaas_ike_policy_v2" "vpn1" {
  ike_version = "v2"
  encryption_algorithm = "aes-256"
  auth_algorithm = "sha2-256"
  pfs = "group16"
  lifetime {
    value = 28800
    units = "seconds"
  }
}

resource "opentelekomcloud_vpnaas_ipsec_policy_v2" "vpn1" {
  encryption_algorithm = "aes-256"
  auth_algorithm = "sha2-256"
  pfs = "group16"
  lifetime {
    value = 3600
    units = "seconds"
  }
}
```

# Debug Output/Panic Output
N/A

# Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. `terraform apply`

# Expected Behavior
What should have happened?
> PFS Group16 is selected and used in the policies.
> Second run of `terraform apply` does not show any differences pending.

# Actual Behavior
What actually happened?
> PFS Group16 is selected but not used in the policies, it falls back to default group5.
> Second run of `terraform apply` does show the differences pending, e.g. group5 -> group16.

# Important Factoids
N/A

# References
Code: https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/blob/v1.20.2/opentelekomcloud/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go#L349
Doc IPsec: https://docs.otc.t-systems.com/api/vpn/en_topic_0093011504.html
Doc IKE: https://docs.otc.t-systems.com/api/vpn/en_topic_0093011510.html
Switch statement is not supporting options:
- group15
- group16
- group19
- group20
- group21

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request
Refactoring ipsec_policy resource

Fixes: #640 

## PR Checklist

* [x] Refers to: #640
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccVpnIPSecPolicyV2_basic
--- PASS: TestAccVpnIPSecPolicyV2_basic (25.15s)
=== RUN   TestAccVpnIPSecPolicyV2_withLifetime
--- PASS: TestAccVpnIPSecPolicyV2_withLifetime (24.11s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go [hunk 1]
```diff
@@ -36,28 +38,38 @@ func ResourceVpnIPSecPolicyV2() *schema.Resource {
...
+				Default:  "sha1",
+				ValidateFunc: validation.StringInSlice([]string{
+					"md5", "sha1", "sha2-256", "sha2-384", "sha2-512",
+				}, false),
 			},
 			"encapsulation_mode": {
 				Type:     schema.TypeString,
 				Optional: true,
-				Computed: true,
+				Default:  "tunnel",
```
Comment (outcatcher) [inline opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go]: Is `encapsulation_mode` not computed?

File: opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go [hunk 2]
```diff
@@ -36,28 +38,38 @@ func ResourceVpnIPSecPolicyV2() *schema.Resource {
...
 			},
 			"pfs": {
 				Type:     schema.TypeString,
 				Optional: true,
-				Computed: true,
+				Default:  "group5",
+				ValidateFunc: validation.StringInSlice([]string{
+					"group1", "group2", "group5", "group14", "group15", "group16", "group19", "group20", "group21",
```
Comment (outcatcher) [inline opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go]: List of groups here and in the doc differs

File: opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go [hunk 3]
```diff
@@ -66,9 +78,11 @@ func ResourceVpnIPSecPolicyV2() *schema.Resource {
 			"transform_protocol": {
 				Type:     schema.TypeString,
 				Optional: true,
-				Computed: true,
+				Default:  "esp",
+				ValidateFunc: validation.StringInSlice([]string{
+					"esp", "ah", "ah-esp",
+				}, false),
```
Comment (outcatcher) [inline opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go]: Why is it not computed any more?

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go (hunk 1 above)
Comment (lego963) [reply opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go]: Cannot be used when Default is specified

File: opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go (hunk 2 above)
Comment (lego963) [reply opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go]: updated

File: opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go (hunk 3 above)
Comment (lego963) [reply opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_ipsec_policy_v2.go]: Cannot be used when Default is specified

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**